xcode/Subconscious/Shared/Components/AddressBook/DidQrCodeView.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  DidQrCodeView.swift
+//  Subconscious (iOS)
+//
+//  Created by Ben Follington on 27/2/2023.
+//
+
+import Foundation
+import SwiftUI
+import CoreImage.CIFilterBuiltins
+
+struct DidQrCodeView: View {
+    var did: Did
+    
+    let context = CIContext()
+    let filter = CIFilter.qrCodeGenerator()
+    let colorInvertFilter = CIFilter.colorInvert()
+    let maskToAlphaFilter = CIFilter.maskToAlpha()
+    let colorReplaceFilter = CIFilter.spotColor()
+    
+    let maskFilter = CIFilter.blendWithMask()
+    
+    func generateQRCode(from string: String) -> UIImage? {
+        filter.message = Data(string.utf8)
+        
+        colorInvertFilter.inputImage = filter.outputImage
+        guard let inverted = colorInvertFilter.outputImage else { return nil }
+        
+        maskFilter.maskImage = inverted
+        // This is the only way I managed to produce a CIColor from .accentColor
+        // https://developer.apple.com/forums/thread/687764
+        // https://stackoverflow.com/questions/56586055/how-to-get-rgb-components-from-color-in-swiftui
+        var r: CGFloat = 0.0
+        var g: CGFloat = 0.0
+        var b: CGFloat = 0.0
+        var a: CGFloat = 0.0
+        UIColor(Color.accentColor).getRed(&r, green: &g, blue: &b, alpha: &a)
+        let col = CIColor(red: r, green: g, blue: b, alpha: a)
+        maskFilter.inputImage = CIImage(color: col)
```

Comment:
Convert `accentColor` to a `CIColor` and then construct solid-color image to blend with the mask we created above.